packages/block-editor/src/components/colors/utils.js

```diff
@@ -63,11 +63,12 @@ export function getColorClassName( colorContextName, colorSlug ) {
 	// We don't want to use kebabCase from lodash here
 	// see https://github.com/WordPress/gutenberg/issues/32347
 	// However, we need to make sure the generated class
-	// doesn't contain spaces.
-	return `has-${ colorSlug.replace(
+	// doesn't contain spaces, or any special characters.
+	const slug = colorSlug.replace( /[^a-zA-Z0-9 ]/g, '' );
```

Comment:
```suggestion
	const slug = colorSlug.replace( /[^a-zA-Z0-9\- ]/g, '' );
```

I believe we need to cover hyphenated color slug names e.g. `subtle-background`. With the current regex the hyphen gets stripped resulting in `subtlebackground`. This is reflected in the e2e errors as well.

Reply:
Given that we're accomodating this code to fix bugs in cases that don't happen in core, I'd appreciate some unit tests for this as well.

Reply:
I also think that third-parties should pass proper data: in this case, it should be a string.